lib/scoped_search/query_builder.rb

```diff
@@ -311,7 +311,7 @@ def to_sql(operator = nil, &block) # :yields: finder_option_type, value
           klass_table_name = relation ? "#{klass.table_name}_#{num}" : klass.table_name
           return "#{connection.quote_table_name(klass_table_name)}.#{connection.quote_column_name(field.to_s)}"
         elsif relation
-          yield(:include, relation)
+          yield(:include, relation) unless definition.reflection_by_name(definition.klass, relation).macro == :has_many && [:eq, :ne, :gt, :gte, :lt, :lte].include?(operator)
```

Comment:
This breaks a bit of encapsulation in this method I think, it's checking what the calling method (i.e. sql_test) is doing in order to change its behaviour. This method shouldn't care about this detail in the relation + operator.

It'd be better for `sql_test` to pass an option or similar to this method when it's generating a subquery to either explicitly add or skip the `:include`, as that's where the logic for subquery vs join is made. (e.g. `field.to_sql(operator, no_include: true, &block)`)